presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -480,14 +482,16 @@ void PrestoServer::run() {
     registerSidecarEndpoints();
   }
 
-  std::string taskUri;
-  if (httpsPort.has_value()) {
-    taskUri = fmt::format(kTaskUriFormat, kHttps, address_, httpsPort.value());
-  } else {
-    taskUri = fmt::format(kTaskUriFormat, kHttp, address_, httpPort);
-  }
+  auto setTaskUri = [&](bool useHttps, int port) {
```

Comment:
Can we move this to where it is first used in the code? thanks!

nit: s/setTaskUri/setTaskUriCb or setTaskUriFn/

Reply:
I'm moving this (and the other lambda) before the httpServer start.

I'm thinking of moving the httpServer lambda into it's own function too if we have to move this inside because things get a bit less readable if we put these (longer) lambdas into the server lamdba as well.